src/files/pyfingerprint/pyfingerprint.py

```diff
@@ -735,22 +730,29 @@ def downloadImage(self, imageDestination):
             if ( receivedPacketType != FINGERPRINT_DATAPACKET and receivedPacketType != FINGERPRINT_ENDDATAPACKET ):
                 raise Exception('The received packet is no data packet!')
 
-            ## X coordinate of current pixel
-            x = 0
-
-            for i in range(0, len(receivedPacketPayload)):
-
-                ## Thanks to Danylo Esterman <soundcracker@gmail.com> for the "multiple with 17" improvement:
+            imageData.append(receivedPacketPayload)
 
-                ## Draw left 4 Bits one byte of package
-                pixels[x, line] = (receivedPacketPayload[i] >> 4) * 17
-                x = x + 1
+        ## Initialize image
+        resultImage = Image.new('L', (256, 288), 'white')
+        pixels = resultImage.load()
+        row = 0
+        column = 0
 
-                ## Draw right 4 Bits one byte of package
-                pixels[x, line] = (receivedPacketPayload[i] & 0b00001111) * 17
-                x = x + 1
+        for y in range(resultImage.height):
+            for x in range(resultImage.width):
 
-            line = line + 1
+                ## One byte contains two pixels
+                ## Thanks to Danylo Esterman <soundcracker@gmail.com> for the "multiple with 17" improvement:
+                if (x % 2 == 0):
+                    pixels[x, y] = (imageData[row][column]  >> 4) * 17
```

Comment:
The sense of the shift operations is not clear without comments. Please re-add the comments.

Reply:
Sure